### PR TITLE
Event based aggregation plus sweeping with threads

### PR DESCRIPTION
**Pull request by meatballhat**
and logging updates, and rubocop-driven cleaning while re-familiarizing, and removing the batch-based async aggregation stuff.

### CODE REVIEW

File: lib/travis/logs/aggregate.rb [hunk 1]
```diff
@@ -19,15 +19,23 @@ def setup
...
       end
 
       def aggregate_logs
-        Travis::Logs::Services::AggregateLogs.run
+        aggregator.run
       rescue Exception => e
         Travis::Exceptions.handle(e)
       end
+
+      private def aggregator
```
Comment (bnferguson) [inline lib/travis/logs/aggregate.rb]: Small style nit, putting `private` in front of individual methods is typically non-idiomatic for ruby which favors them done as a group ala:

```
private 

def aggregator 
......
end

def sleep_interval
.....
end
```

**Comment by bnferguson** [comment]
I know you're responding to an incident and having to roll fast on this stuff (:heart:), but when you get a chance, do you mind filling in the description more for the thinking behind "Add thread pool executor to aggregation", the problem and what this solves? This'll be big for me understanding it to review as well as understanding how it may affect how we do things in Enterprise (esp since we run lock aggregate runs - or will once I get that bit load tested more :D).

File: lib/travis/logs/aggregate.rb (hunk 1 above)
Comment (bnferguson) [inline lib/travis/logs/aggregate.rb]: (unless we've started doing it more inline these days? I just haven't seen it in most of our code, though I know there can be some good reasons behind it - clean code style etc)

**Comment by svenfuchs** [comment]
@bnferguson it might be more of an intermediate emergency solution, i guess

longterm i think it would be good to remove the loop that checks for aggregatable log ids, and instead trigger/schedule aggregation when we receive the `final` log part. that would make the whole thing more scalable (as far as i understand we can still only run a single aggregation process at the moment).

**Comment by bnferguson** [comment]
@svenfuchs That makes sense! Mostly trying to understand how this works and how it may affect Enterprise installs if things go a little weird. 

I really like your plan for the future, totally agree that it'd make things a lot more scalable. Also remove the need for locking in HA setups. 💃

File: lib/travis/logs/aggregate.rb (hunk 1 above)
Comment (meatballhat) [inline lib/travis/logs/aggregate.rb]: I'm following @rkh's lead on this, ja

Comment (bnferguson) [inline lib/travis/logs/aggregate.rb]: > I'm following @rkh's lead on this, ja

Right on! I hadn't seen this yet so wanted to make sure!

Comment (rkh) [inline lib/travis/logs/aggregate.rb]: It's a style I started experimenting it and I kinda like it, as you can see from looking at the method that it's private without having to scan the entire class and don't accidentally add methods with the wrong visibility (I think that Sven indents private methods one level further for the same reason, but I personally dislike this and it makes you still scan for indentation mismatch).

**Comment by bnferguson** [comment]
@meatballhat @svenfuchs - With this merged and things going event based, is https://github.com/travis-ci/travis-logs/pull/70 even needed anymore for future Enterprise releases branched off of master?

**Pull request merged**